Projects/Server/Utilities/Random.cs

```diff
@@ -49,7 +48,10 @@ static RandomImpl()
 
   public interface IRandomImpl
   {
-    int Next(int c);
+    uint NextUInt32();
+    uint Next(uint n);

```

Comment:
I think you also need a separate crypto random that can be used for passwords and such. Maybe have two random implementations.

Reply:
Also, since the crypto random is only used for passwords and isn't in the performance path, can you just use the one in the standard library directly for the password code and then entirely eliminate the use of rdrand? That would simplify a ton